src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/time.html.twig

```diff
@@ -41,63 +41,82 @@
-<span class="label">
-    <span class="icon"><svg width="30" height="33" xmlns="http://www.w3.org/2000/svg" version="1.1" x="0px" y="0px" viewBox="0 0 30 33" enable-background="new 0 0 30 33" xml:space="preserve"><rect x="11" y="2" fill="#3F3F3F" width="8" height="3"/><path fill="#3F3F3F" d="M15 6.1c-6.6 0-12 5.4-12 12s5.4 12 12 12s12-5.4 12-12S21.6 6.1 15 6.1z M22 20h-6h-3V10h3v7h6V20z"/></svg></span>
-    <strong>Timeline</strong>
-</span>
+    <span class="label">
+        <span class="icon">{{ include('@WebProfiler/Icon/time.svg') }}</span>
+        <strong>Performance</strong>
+    </span>
 {% endblock %}
 
 {% block panel %}
-    <h2>Timeline</h2>
-    {% if collector.events|length %}
-        {{ block('panelContent') }}
+    {% if collector.events is empty %}
+        <div class="empty">
+            <p>No timing events have been recorded. Are you sure that debugging is enabled in the kernel?</p>
+        </div>
     {% else %}
-        <p>
-            <em>No timing events have been recorded. Are you sure that debugging is enabled in the kernel?</em>
-        </p>
+        <h2>Performance metrics</h2>
+
+        <div class="metrics">
+            <div class="metric">
+                <span class="value">{{ '%.0f'|format(collector.duration) }} <span class="unit">ms</span></span>
+                <span class="label">Total execution time</span>
+            </div>
+
+            <div class="metric">
+                <span class="value">{{ '%.0f'|format(collector.inittime) }} <span class="unit">ms</span></span>
+                <span class="label">Symfony initialization</span>
+            </div>
+
+            {% if profile.collectors.memory %}
+                <div class="metric">
+                    <span class="value">{{ '%.2f'|format(profile.collectors.memory.memory / 1024 / 1024) }} <span class="unit">MB</span></span>
+                    <span class="label">Peak memory usage</span>
+                </div>
+            {% endif %}
+        </div>
+
+        <h2>Execution timeline</h2>
+
+        {{ block('panelContent') }}
     {% endif %}
 {% endblock %}
 
 {% block panelContent %}
     <form id="timeline-control" action="" method="get">
         <input type="hidden" name="panel" value="time">
-        <table>
-            <tr>
-                <th style="width: 20%">Total time</th>
-                <td>{{ '%.0f'|format(collector.duration) }} ms</td>
-            </tr>
-            <tr>
-                <th>Initialization time</th>
-                <td>{{ '%.0f'|format(collector.inittime) }} ms</td>
-            </tr>
-            <tr>
-                <th>Threshold</th>
-                <td><input type="number" size="3" name="threshold" value="1" min="0"> ms</td>
-            </tr>
-        </table>
+        <label for="threshold">Threshold</label>
+        <input type="number" size="3" name="threshold" value="3" min="0"> ms
+        <span class="help">(timeline only displays events slower than this threshold)</span>
```

Comment:
There is a difference between slower and longer: If an event is *slower* than another one, it takes *longer* until it is finished.

In this case, I think slower and longer can both be used, but longer seems to be better (not a native though, /cc @weaverryan )